scylla-cdc/src/lib.rs

```diff
@@ -5,9 +5,426 @@ pub mod stream_generations;
+        async fn consume_cdc(&mut self, mut data: CDCRow<'_>) -> anyhow::Result<()> {
+            let pk_val = if data.column_exists("pk") {
+                PrimaryKeyValue::from_cql(data.get_value("pk").as_ref().unwrap()).unwrap()
+            } else {
+                // If primary key is composed of many columns, they will have names pk1, pk2...
+                let mut values = Vec::with_capacity(MAX_PK_SIZE);
+                let mut i = 1;
+                while data.column_exists(&format!("pk{}", i)) {
+                    let val = data.get_value(&format!("pk{}", i)).as_ref().unwrap();
+                    values.push(PrimaryKeyValue::from_cql(val).unwrap());
+                    i += 1;
+                }
+
+                PrimaryKeyValue::List(values)
+            };
+            let op_type = data.operation.clone();
+            let ck = data.take_value("ck").unwrap().as_int().unwrap();
+            let val = data.take_value("v").map(|cql| cql.as_int().unwrap());
+
+            self.read_operations
+                .lock()
+                .await
+                .entry(pk_val)
+                .or_insert_with(VecDeque::new)
+                .push_back(Operation::new(op_type, ck, val));
+
+            Ok(())
+        }
+    }
+
+    struct TestConsumerFactory {
+        read_operations: Arc<Mutex<HashMap<PrimaryKeyValue, VecDeque<Operation>>>>,
+    }
+
+    impl TestConsumerFactory {
+        fn new(
+            operations: Arc<Mutex<HashMap<PrimaryKeyValue, VecDeque<Operation>>>>,
+        ) -> TestConsumerFactory {
+            TestConsumerFactory {
+                read_operations: operations,
+            }
+        }
+    }
+
+    #[async_trait]
+    impl ConsumerFactory for TestConsumerFactory {
+        async fn new_consumer(&self) -> Box<dyn Consumer> {
+            Box::new(TestConsumer {
+                read_operations: Arc::clone(&self.read_operations),
+            })
+        }
+    }
+
+    fn get_uri() -> String {
+        std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string())
+    }
+
+    async fn get_session() -> anyhow::Result<Session> {
+        Ok(SessionBuilder::new().known_node(get_uri()).build().await?)
+    }
+
+    // Creates test table and keyspace. Query should not contain keyspace name, only the table name.
+    async fn create_table_and_keyspace(
+        session: &Session,
+        table_name: &str,
+        pk_type_names: Vec<&str>,
+    ) -> anyhow::Result<(PreparedStatement, PreparedStatement)> {
+        session.query("CREATE KEYSPACE IF NOT EXISTS e2e_test WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }", ()).await?;
+        session.use_keyspace("e2e_test", false).await?;
+
+        session
+            .query(format!("DROP TABLE IF EXISTS {}", table_name), ())
+            .await?;
+
+        let (create_query, insert_query, update_query) = if pk_type_names.len() == 1 {
+            (
+                format!(
```

Comment:
For the sake of simplicity, let's get rid of the special case for `pk_type_names.len() == 1` and just use the `pk1` name in such case.